*** maze_v2.c	2023-11-25 03:36:21.626440000 +0300
--- maze_v3.c	2023-11-25 03:39:06.105568300 +0300
***************
*** 93,104 ****
  int main(int argc, char **argv) {
      srand(time(0));
      
!     if (argc != 2) {
!         printf("Usage: ./maze maze_size\n");
          return 1;
      }
  
!     int maze_size = atoi(argv[1]) * 2 + 1;
      int **maze  = malloc(maze_size * sizeof(int*));
  
      for (int i = 0; i < maze_size; ++i) {
--- 93,104 ----
  int main(int argc, char **argv) {
      srand(time(0));
      
!     if (argc != 3) {
!         printf("Usage: ./maze <'door''wall'> maze_size\n");
          return 1;
      }
  
!     int maze_size = atoi(argv[2]) * 2 + 1;
      int **maze  = malloc(maze_size * sizeof(int*));
  
      for (int i = 0; i < maze_size; ++i) {
***************
*** 159,167 ****
      for (int i = 0; i < maze_size; ++i) {
          for (int j = 0; j < maze_size; ++j) {
              if (maze[i][j] == WALL) {
!                 printf("%c", WALL_SYM);
              } else {
!                 printf("%c", DOOR_SYM);
              }
          }
          printf("\n");
--- 159,167 ----
      for (int i = 0; i < maze_size; ++i) {
          for (int j = 0; j < maze_size; ++j) {
              if (maze[i][j] == WALL) {
!                 printf("%c", argv[1][1]);
              } else {
!                 printf("%c", argv[1][0]);
              }
          }
          printf("\n");
